cmd/observer/node_builder/observer_builder.go

```diff
@@ -175,6 +180,25 @@ func (builder *ObserverServiceBuilder) deriveBootstrapPeerIdentities() error {
 	return nil
 }
 
+// deriveBootstrapPeerIdentities derives the Flow Identity of the bootstrap peers from the parameters.
+// These are the identities of the observers also acting as the DHT bootstrap server
+func (builder *ObserverServiceBuilder) deriveUpstreamPeerIdentities() error {
+	// if bootstrap identities already provided (as part of alternate initialization as a library the skip reading command
+	// line params)
+	if builder.upstreamIdentities != nil {
+		return nil
+	}
+
+	ids, err := BootstrapIdentities(builder.upstreamNodeAddresses, builder.upstreamNodePublicKeys)
```

Comment:
`BootstrapIdentities` requires all addresses have a valid public key. Ideally we don't require proxying over TLS. Maybe if the public key list is empty, you just create a list of identities that includes only addresses?

Reply:
Fixed: https://github.com/GetElastech/flow-go/commit/e867cb36f4fde512a7b0cd4d15f6755a46473939